### DISCUSSION TITLE
Document and validate paging parameters and behavior

### DISCUSSION THREAD

**Discussion by ITDancer13** [Biarity/Sieve#137] [Ideas]
I'd suggest to rethink our pagination. At the moment it's a bit confusing having nullable integers on our model with 1...int.MaxValue restrictions which are never checked. Additionally we have default values as configuration option and internal logic like `if (pageSize <= 0)` paging will not be applied.

My suggestion for version 3 would be to define and document the following restrictions:

- [ ] Valid values for `PageSize `and `DefaultPageSize`: 0...int.MaxValue with 0 = paging disabled
- [ ] Valid values `Page`: 1...int.MaxValue

If invalid values are passed to `SieveProcessor.ApplyPagination` an specific exception should be thrown.

**Comment by hasanmanzak** [comment]
I overriden the behaviour of paging as `if requested page is -1 then disable paging` as a wrap around kind of thought, not the `pageSize`.
